src/appleseed/renderer/kernel/lighting/directlightingintegrator.cpp

```diff
@@ -670,13 +676,32 @@ void DirectLightingIntegrator::add_non_physical_light_sample_contribution(
     // Evaluate the light.
     Vector3d emission_position, emission_direction;
     Spectrum light_value(Spectrum::Illuminance);
-    light->evaluate(
+
+    // Generate a uniform sample in [0,1).
+    sampling_context.split_in_place(2, 1);
+    const Vector2d s = sampling_context.next2<Vector2d>();
+    float probability;
+
+    light->sample(
         m_shading_context,
         sample.m_light_transform,
         m_material_sampler.get_point(),
+        s,
         emission_position,
         emission_direction,
-        light_value);
+        light_value,
+        probability
+    );
+
+    //sample.m_probability *= probability;
```

Comment:
I've tried to apply calculated probability to light sample but it made light many times brighter.